data_schema/convert_value.py

```diff
@@ -154,6 +154,11 @@ def _convert_value(self, value, format_str):
         if self.is_string(value):
             value = datetime.strptime(value, format_str) if format_str else parse(value)
 
+        # It is assumed that value is a datetime here. If it isn't a datetime, then it is a bad value like
+        # a number that is too large to be parsed as an integer
+        if type(value) != datetime:
```

Comment:
it is using `datetime.utcfromtimestamp(float(value))` so it will be a datetime. the date converter uses the datetime converter and datetime type

```
FieldSchemaType.DATE: DatetimeConverter(FieldSchemaType.DATE, datetime)
```
that is where it is defined saying our DATE format uses the `DatetimeConverter` and converts it to a `datetime` object when parsing